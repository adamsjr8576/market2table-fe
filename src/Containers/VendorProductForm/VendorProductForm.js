import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './VendorProductForm.scss';

export const VendorProductForm = () => {

  const [productName, setProductName] = useState('');
  const [productPrice, setProductPrice] = useState('');
  const [productDescription, setProductDescription] = useState('');
  const [hasError, setHasError] = useState(false);

  return (
    <form className='vendor-product-form'>
      <section className='vendor-product-from-section'>
        <input value={productName} onChange={(event) => setProductName(event.target.value)} type='text' placeholder='Product Name...' className='product-name-input' maxlength='25'/>
        <input value={productPrice} onChange={(event) => setProductPrice(event.target.value)}type='number' placeholder='Price...' className='product-price-input'/>
      </section>
      <textarea value={productDescription} onChange={(event) => setProductDescription(event.target.value)}
        form='vendor-form-info' name='vendor-description-textarea'
        className='product-description-textarea' placeholder='Product Description...'
        rows='3' columns='25' maxlength='65'>
      </textarea>
      <button type='button' className='submit-new-product-button'>Create Product</button>
    </form>
  )
}

export default VendorProductForm;
