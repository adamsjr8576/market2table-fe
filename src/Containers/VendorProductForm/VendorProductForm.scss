input[type=number]::-webkit-inner-spin-button,
input[type=number]::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.vendor-product-form {
  width: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 175px;
}

.vendor-product-from-section {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.product-name-input {
  border-radius: 20px;
  border: 2px solid black;
  background-color: white;
  padding: 10px;
  outline: 0;
  font-size: 1em;
  width: 50%;
}

.product-price-input {
  border-radius: 20px;
  border: 2px solid black;
  background-color: white;
  padding: 10px;
  outline: 0;
  font-size: 1em;
  width: 20%;
}

.product-description-textarea {
  resize: none;
  font-size: 1em;
  outline: 0;
  border: 2px solid black;
  border-radius: 10px;
  padding: 10px;
  width: 60%;
}

.submit-new-product-button {
  width: 35%;
  min-width: 130px;
  padding: 5px;
  font-size: 1em;
  border: 2px solid #3B3D39;
  background-color: #F0F0EF;
  outline: 0;
  border-radius: 10px;
  cursor: pointer;
}

.submit-new-product-button:hover,
.submit-new-product-button:focus {
  background-color: #C4D1B2;
  border: 2px solid #5E7838;
}

.submit-new-product-button:active {
  transform: scale(.98);
}
