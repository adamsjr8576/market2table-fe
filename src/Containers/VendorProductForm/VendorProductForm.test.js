import React from 'react';
import { shallow } from 'enzyme';
import VendorProductForm from './VendorProductForm';
import { useSelector, useDispatch } from 'react-redux';


describe('VendorProductForm', () => {
    let mockproducts = [{ name: 'apples', description: 'lots', price: 1.99 }, { name: 'oranges', description: 'lots', price: 4 }]
    let mockSetProducts = jest.fn();
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<VendorProductForm
        setProducts={mockSetProducts}
        products={mockproducts}
      />);
      const mockMath = Object.create(global.Date);
      mockMath.now = () => 15;
      global.Date = mockMath;
    });

    it('should match the VendorProductForm snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should have a default value of '' for the product name input", () => {
      expect(wrapper.find('.product-name-input').prop('value')).toEqual('');
    });

    it("should update productName state on change of product name input", () => {
      wrapper.find('.product-name-input').simulate('change', {target: {value: 'Celery'}})
      expect(wrapper.find('.product-name-input').prop('value')).toEqual('Celery');
    });

    it("should have a default value of '' for the product price input", () => {
      expect(wrapper.find('.product-price-input').prop('value')).toEqual('');
    });

    it("should update productPrice state on change of product price input", () => {
      wrapper.find('.product-price-input').simulate('change', {target: {value: '5'}});
      expect(wrapper.find('.product-price-input').prop('value')).toEqual('5');
    });

    it("should have a default value of '' for the product description textarea", () => {
      expect(wrapper.find('.product-description-textarea').prop('value')).toEqual('');
    });

    it("should update productPrice state on change of product price input", () => {
      wrapper.find('.product-description-textarea').simulate('change', {target: {value: 'The crunchiest celery around'}});
      expect(wrapper.find('.product-description-textarea').prop('value')).toEqual('The crunchiest celery around');
    });

    it("should invoke setproducts on button click with new product and current products", () => {
      wrapper.find('.product-name-input').simulate('change', {target: {value: 'Celery'}});
      wrapper.find('.product-price-input').simulate('change', {target: {value: '5'}});
      wrapper.find('.product-description-textarea').simulate('change', {target: {value: 'The crunchiest celery around'}});
      wrapper.find('.submit-new-product-button').simulate('click');
      expect(mockSetProducts).toHaveBeenCalledWith([...mockproducts, {name: 'Celery', price: '5', description: 'The crunchiest celery around', id: 15}])
    });

    it("should invoke setproducts on button click with new product and current products", () => {
      wrapper.find('.product-name-input').simulate('change', {target: {value: 'Celery'}});
      wrapper.find('.product-price-input').simulate('change', {target: {value: '5'}});
      wrapper.find('.product-description-textarea').simulate('change', {target: {value: 'The crunchiest celery around'}});
      wrapper.find('.submit-new-product-button').simulate('click');
      expect(wrapper.find('.product-description-textarea').prop('value')).toEqual('');
      expect(wrapper.find('.product-price-input').prop('value')).toEqual('');
      expect(wrapper.find('.product-name-input').prop('value')).toEqual('');
    });
});
