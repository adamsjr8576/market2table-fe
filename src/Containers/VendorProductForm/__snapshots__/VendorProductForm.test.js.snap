// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorProductForm should match the VendorProductForm snapshot 1`] = `
<form
  className="vendor-product-form"
>
  <section
    className="vendor-product-from-section"
  >
    <input
      className="product-name-input"
      maxlength="20"
      onChange={[Function]}
      placeholder="Product Name..."
      type="text"
      value=""
    />
    <input
      className="product-price-input"
      maxlength="10"
      onChange={[Function]}
      placeholder="$/qty..."
      type="text"
      value=""
    />
  </section>
  <textarea
    className="product-description-textarea"
    columns="25"
    form="vendor-form-info"
    maxlength="65"
    name="vendor-description-textarea"
    onChange={[Function]}
    placeholder="Product Description..."
    rows="3"
    value=""
  />
  <button
    className="submit-new-product-button"
    onClick={[Function]}
    type="button"
  >
    Create Product
  </button>
</form>
`;
